hack/golangci.yaml.in

```diff
@@ -206,3 +209,18 @@ linters-settings: # please keep this alphabetized
     checks:
       - "ST1019"   # Importing the same package multiple times
   {{- end}}
+  {{- if .Strict}}
+  testifylint:
+    enable-all: true
+    disable:
```

Comment:
Great idea to enable this linter, thanks! We did something similar for gomega+ginkgo before and it helps write better tests.

About the disabled checks: can you add a comment *why* they are disabled?

You updated existing code, which is good because developers tend to use copy-and-paste, but whouldn't have been absolutely necessary because strict checks only apply to modified code.

So if you only disabled them because it was too much work to update existing code, then we could leave them enabled and rely on developers who touch the code to comply with the stricter checking.

Reply:
That's what I meant: you *can* activate the checks in the "strict" config without fixing all existing code.

That's because they are not enabled in the "base" configuration, so `hack/verify-golang-ci.sh` passes for existing, unfixed tests.

Reply:
> open more PR for each other linter

Instead of one PR for each linter, doing one PR per subsystem which fixes all known issues is easier to merge. PRs that touch a lot of different areas are difficult to get approved, which might delay them and increase the risk of merge conflicts while they are pending.

How granular such PRs should be is a bit subjective - use common sense and perhaps consider OWNERS files.

Reply:
Why not both?  One PR per subsystem, one commit per linter.

Reply:
:+1: 

I chose to let tide squash commits because @mmorel-35 said that they currently only have access to a phone.

Usually I care a bit more about commit hygiene. @mmorel-35: see https://www.kubernetes.dev/docs/guide/contributing/#best-practices and https://www.kubernetes.dev/docs/guide/pull-requests/#commit-message-guidelines